CanvasColorSpaceProposal.md

```diff
@@ -67,40 +64,48 @@ partial interface CanvasRenderingContext2D {
-* toDataURL/toBlob produce resources tagged as being in the sRGB color space, if the encoding format supports colorspace tagging or embedded color profiles. Exception: User agents that implement the legacy behavior must not encode any color space metadata.
-* Images with no color profile, when drawn to the canvas, are assumed to already be in the canvas's color space, and require no color transformation.
+* Displayed canvases must be converted to the display's color profile if available. This happens downstream at the compositing stage, and has no script-visible side-effects.
 
-#### The "rec2020" color space
+#### The "extended-srgb" color space
 * Support is optional.
-* User agents that select the rec2020 color gamut in CSS media queries must support this color space.
-* The color space corresponds to ITU-R Recommendation BT.2020.
-* toDataURL/toBlob convert image data to the rec-2020 color space.
-* Images with no color profile, when drawn to the canvas, are assumed to be in the sRGB color space, and are converted to rec2020 for the purpose of the draw.
-
-#### The "p3" color space
+* Requires that the pixelFormat attribute be "float16".
+* This color space exists to enable existing applications to add wide color gamut and high dynamic range support with no little to no modification
+* Color space definition
+   * This color space uses the same primaries as the sRGB color space.
+   * This color space supports arbitrary pixel values, in particular, values outside of the range [0, 1].
+   * Pixel values greater than 1 are defined by extending nonlinear piece of the sRGB transfer function.
+   * Pixel values less than 0 are defined by point-symmetrically extending the sRGB transfer function. Specifically, T(-x)=sign(x) * T(abs(x))
+
+#### The "linear-srgb" color space
 * Support is optional.
-* User agents that select either of the the p3 or the rec2020 color gamuts in CSS media queries must support this color space.
-* The color space corresponds to the DCI P3 specification (TODO: add link).
-* toDataURL/toBlob convert image data to the p3 space.
-* Images with no color profile, when drawn to the canvas, are assumed to be in the sRGB color space, and are converted to p3 for the purpose of the draw.
+* Requires that the pixelFormat attribute be "float16".
+* This color space exists to enable wide color gamut and high dynamic range applications that use physically correct linear color operations (e.g, blending and interpolation)
+* Color space definition
+   * This color space uses the same primaries as the sRGB color space.
+   * This color space has a linear transfer function that supports arbitrary pixel values, in particular, values outside of the range [0, 1].
+* Note that gradients will be physically uniform, and thus will not match gradients in an "srgb" canvas
+* Note that pixel values written by WebGL shaders are now in a space with a linear transfer function, and will not necessarily match those of an "srgb" canvas.
```

Comment:
Maybe for a future PR: I think we need to provide an explainer paragraph to explain why we don't need additional color space support (other primaries), as far as displaying is concerned. And we should provide an example of how to hack around it to use linear-srgb mode to composite and filter images in an intermediate working space defined by arbitrary primaries. We only need to convert the final output from the intermediate space to linear-srgb so that the colors are interpreted correctly on scan-out.

Reply:
Yes, we should put something in there.